
version: '3.8'

services:
  envoy:
    image: envoyproxy/envoy-alpine:v1.14-latest
    container_name: envoy
    volumes:
      - ./envoy.yaml:/etc/envoy/envoy.yaml
    ports:
      - "8080:8080"
      - "9901:9901"
    command: ["envoy", "-c", "/etc/envoy/envoy.yaml"]
    networks:
      - auth_network
  auth:
    build:
      context: ./auth
    depends_on:
      - auth_db
      - notification
    ports:
      - "4000:4000"
    networks:
      - auth_network
  notification:
    build:
      context: ./notification
    depends_on:
      - notification_db
    ports:
      - "3000:3000"
    networks:
      - auth_network
  auth_db:
    image: postgres
    env_file:
       - ./auth/.env
    ports:
      - "5432:5432"
    volumes:
      - pgdata2:/var/lib/postgresql/data
      - ./auth/migrations:/docker-entrypoint-initdb.d
    networks:
      - auth_network
  notification_db:
    image: mongo:latest
    ports:
      - "27017:27017"
    volumes:
      - mongodb_data:/data/db
    env_file:
       - ./notification/.env
    networks:
      - auth_network
volumes:
  pgdata2:
  mongodb_data:
networks:
  auth_network:
    driver: bridge
